pulser-core/pulser/sequence/sequence.py

```diff
@@ -376,74 +377,130 @@ def switch_device(
-
-                        if strict_error_message is None:
-                            sw_channel_args = list(call.args)
-                            # Switch the old id with the correct id
-                            sw_channel_args[1] = nd_ch_id
-                            getattr(new_seq, "declare_channel")(
-                                *sw_channel_args, **call.kwargs
-                            )
-                            called = True
-                            break
-                    elif basis_match and not addressing_match:
-                        ch_type_er_mess = "No addressing match."
-                    elif not basis_match and addressing_match:
-                        ch_type_er_mess = "No basis match."
+                                matched = False
+                                strict_error_message = (
+                                    "No channel with phase_jump_time"
+                                    + " & clock_period match."
+                                )
                     else:
-                        ch_type_er_mess = "No basis and addressing match."
+                        channel_matching[channel_name] = nd_ch_id
+                        matched = True
+                        break
+                else:
+                    matched = False
+                    ch_type_er_mess = (
+                        "No channel with both basis and addressing match."
+                    )
 
-                if not called:
-                    if strict_error_message is not None:
-                        raise ValueError(strict_error_message)
-                    else:
-                        raise TypeError(ch_type_er_mess)
+            if not matched:
+                if strict_error_message is not None:
+                    raise ValueError(strict_error_message)
+                else:
+                    raise TypeError(ch_type_er_mess)
+
+        for call in self._calls[1:]:
+            if call.name == "declare_channel":
+                sw_channel_args = list(call.args)
+                # Switch the old id with the correct id
+                sw_channel_args[1] = channel_matching[sw_channel_args[0]]
+                if strict and alert1:
+                    warnings.warn(
+                        "The phase_jump_time of the new device"
+                        + " is different, take it in account"
+                        + " for the upcoming pulses.",
+                        stacklevel=2,
+                    )
+                getattr(new_seq, "declare_channel")(
+                    *sw_channel_args, **call.kwargs
+                )
             else:
                 getattr(new_seq, call.name)(*call.args, **call.kwargs)
+
+        if new_device.rydberg_level != self._device.rydberg_level:
+            warnings.warn(
+                "The rydberg level of the new device is different"
+                + " take in account in your computations.",
```

Comment:
Instead of the `take in account in your computations` part, perhaps it's more useful to say what the old and new Rydberg levels are, eg
```suggestion
                f"The rydberg level of the new device ({new_device.rydberg_level})"
                f" is different from the current device's ({self._device.rydberg_level}).",
```
Also, I'm thinking that this should also be an error instead of a warning if `strict=True` and any of the new channels is a `Rydberg` channel.